src/Repeater/DefaultRepeater.ts

```diff
@@ -0,0 +1,147 @@
+import { logger } from '../Utils';
+import {
+  Repeater,
+  RepeaterEventHandler,
+  RepeaterDeployedEvent,
+  RepeaterEvents
+} from './Repeater';
+import { inject, injectable } from 'tsyringe';
+import io, { Socket } from 'socket.io-client';
+import parser from 'socket.io-msgpack-parser';
+import { hostname } from 'os';
+import { once } from 'events';
+
+export interface DefaultRepeaterOptions {
+  readonly uri: string;
+  readonly token: string;
+}
+
+export const DefaultRepeaterOptions: unique symbol = Symbol(
+  'DefaultRepeaterOptions'
+);
+
+@injectable()
+export class DefaultRepeater implements Repeater {
+  private latestReconnectionError?: Error;
+  private readonly DEFAULT_RECONNECT_TIMES = 3;
+  private socket?: Socket;
+
+  constructor(
+    @inject(DefaultRepeaterOptions)
+    private readonly options: DefaultRepeaterOptions
+  ) {}
+
+  public disconnect() {
+    if (!this.socket) {
+      throw new Error('Not connected');
+    }
+
+    this.socket?.disconnect();
+    this.socket?.removeAllListeners();
+    this.socket = undefined;
+  }
+
+  public async deploy(
+    repeaterId?: string
+  ): Promise<RepeaterDeployedEvent['response']> {
+    if (!this.socket) {
+      throw new Error('Repeater is not connected yet');
+    }
+
+    this.socket.emit('deploy', {
+      repeaterId
+    });
+
+    const [result]: RepeaterDeployedEvent['response'][] = await once(
+      this.socket,
+      'deployed'
+    );
+
+    return result;
+  }
+
+  public connect() {
+    this.socket = io(this.options.uri, {
+      parser,
+      path: '/api/ws/v1',
+      transports: ['websocket'],
+      reconnectionAttempts: this.DEFAULT_RECONNECT_TIMES,
+      auth: {
+        token: this.options.token,
+        domain: hostname()
```

Comment:
Previously, it was possible to launch multiple repeaters on the same machine. However, the repeater service now restricts this capability to allow only one repeater per workstation. To maintain the existing functionality, you have to use the `repeaterId` instead of the `hostname`.

Reply:
Added. But what about cases when custom  repeater id is not provided? Should we also allow to run multiple repeaters at the same tine in such case?

Reply:
I suppose we should keep the repeater ID option mandatory, at least for now, unless we receive other requirements stating otherwise.